keps/sig-auth/NNNN-20201209-client-exec-proxy/README.md

```diff
@@ -0,0 +1,711 @@
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Proxy Proposal
+
+The request signing proxy is a long-running proxy process which is executed by the client on first use.  This proposal would add configuration options to allow an executed binary to return a localhost port or UDS socket address, over which the client would proxy all API server bound requests.  An additional configuration section for the kubeconfig defines a proxy binary location.
+
+Questions to answer:
+
+1. What is the lifetime of the proxy plugin?
+
+    The proxy plugin process will be executed from a goroutine (executor) the first time it is needed, i.e. on transport setup.  The executor goroutine will complete when the exec command completes.  The client-go library will have an additional work queue.  If the client fails to connect to the socket because no one is listening, an item will be added to the work queue.  An additional goroutine worker will be waiting for work items, and will spawn a new executor goroutine.
```

Comment:
None of this makes sense to be.  The exec plugin is the proxy.

Reply:
Yeah, the exec plugin is (or can be) the proxy.  The only thing that matters is that the proxy is listening on the expected listener after the exec command is run.  The above description is leaving it optional -- the exec plugin can either somehow start the proxy and return, or it can be the proxy and not return, except in error.  In the case where the proxy stop listening, an item will be added to a queue which will trigger a restart attempt of the exec plugin.